src/providers/ad/ad_common.c

```diff
@@ -543,7 +544,7 @@ ad_get_common_options(TALLOC_CTX *mem_ctx,
                       "The hostname [%s] has been expanded to FQDN [%s]. "
                       "If sssd should really use the short hostname, please "
                       "set ad_hostname explicitly.\n", hostname, fqdn);
-                strncpy(hostname, fqdn, sizeof(hostname));
+                strncpy(hostname, fqdn, sizeof(hostname) - 1);
```

Comment:
I think this should be unified:
Use "sizeof(hostname) - 1" in both lines or "HOST_NAME_MAX" in both lines.

Reply:
Done.